client/hooks/auth-context.tsx

```diff
@@ -0,0 +1,33 @@
+import { createContext, ReactElement, ReactNode, useContext, useEffect, useState } from 'react';
+import { UserToken } from '../../common/api-types';
+import { getUser } from '../services/auth-service';
+
+type AuthContextValue = [UserToken | null, (user: UserToken | null) => void];
+
+const AuthContext = createContext<AuthContextValue>([null, () => {}]);
+
+export function AuthContextProvider({ children }: { children: ReactNode }): ReactElement {
+  const [user, setUser] = useState<UserToken | null>(null);
+  const [responseReceived, setResponseReceived] = useState<boolean>(false);
+
+  useEffect(() => {
+    getUser().then(user => {
+      setUser(user);
+      setResponseReceived(true);
+    });
+  }, []);
+
+  if (!responseReceived) {
+    return <></>;
```

Comment:
Have you considered displaying some sort of loading state here rather than an empty fragment? I feel like we might have talked about this last week but I'm not sure